tgui-next/packages/tgui/interfaces/BlackmarketDownlink.js

```diff
@@ -0,0 +1,131 @@
+
+export const BlackmarketDownlink = props => {
+  const { state } = props;
+  const { config, data } = state;
+  const { ref } = config;
+  const categories = data.categories || [];
+  const deliveryMethods = data.delivery_methods || [];
+  const deliveryMethodDesc = data.delivery_method_description || [];
+  const markets = data.markets || {};
+  const items = data.items || {};
+
+  const shipmentSelector = !!data.buying && (
+    <Dimmer
+      textAlign="center">
+      <Grid mt={20}>
+        {map(deliveryMethod => {
+          const name = deliveryMethod.name;
+          if (!(name === "LTSRBT" && !data.ltsrbt_built)) {
+            return (
+              <Grid.Column
+                textAlign="center"
+                position="relative">
+                <Box>
+                  <Box
+                    fontSize="30px">
+                    {name}
+                  </Box>
+                  <Box mt={1}>
+                    {deliveryMethodDesc[name]}
+                  </Box>
+                </Box>
+                <Button
+                  key={name}
+                  content={deliveryMethod.price+'$'}
+                  mt={1}
+                  onClick={() => act(ref, 'buy', {
+                    method: name,
+                  })} />
+              </Grid.Column>
+            );
+          }
+        })(deliveryMethods)}
+      </Grid>
+      <Button
+        content="Cancel"
+        color="bad"
+        onClick={() => act(ref, 'cancel')} />
+    </Dimmer>
+  );
+
+  return (
+    <Fragment>
+      {shipmentSelector}
+      <Section
+        title="Black Market Downlink"
+        buttons={(
+          <Box inline bold>
+            <AnimatedNumber value={Math.round(data.money)} /> credits
+          </Box>
+        )} />
+      <Tabs
+        activeTab={data.viewing_market}>
+        {map(market => {
+          //
+          const id = market.id;
+          const name = market.name;
+          return (
+            <Tabs.Tab
+              key={id}
+              label={name}
+              onClick={() => act(ref, 'set_market', {
+                market: id,
+              })}>
+            </Tabs.Tab>
+          );
+        })(markets)}
+      </Tabs>
+      <Box>
+        <Tabs vertical
+          activeTab={data.viewing_category}>
+          {categories.map(category => (
+            <Tabs.Tab
+              key={category}
+              label={category}
+              height={4}
+              mt={0.5}
+              onClick={() => act(ref, 'set_category', {
+                category: category,
+              })}>
+              {items.map(item => (
+                <Table
+                  key={item.name}
+                  mt={1}
+                  className="candystripe">
+                  <Table.Row>
+                    <Table.Cell bold>
+                      {item.name}
+                    </Table.Cell>
+                    <Table.Cell className="LabeledList__buttons">
```

Comment:
You can use `<Table.Cell collapsing textAlign="right">` for the same effect without sneaking into private API. Check if this has the correct appearance though.

Reply:
This is what cargo uses so I figured it would be fine, but I will try that.

Reply:
I didn't know Cargo used this, and I was planning to refactor LabeledList to not use tables because of some bad side-effects, so I could be breaking a lot of UIs which use these classes directly.

Reply:
Works fine, did that.